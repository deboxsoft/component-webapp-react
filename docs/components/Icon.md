<b>Examples:</b>

1. Icon Sizes:

```js
<div className="icon-wrapper">
    <Icon
        name="smile-o"
        size="xsmall"
    />
    <Icon
        name="smile-o"
        size="small"
    />
    <Icon
        name="smile-o"
        size="medium"
    />
    <Icon
        name="smile-o"
        size="large"
    />
    <Icon
        name="smile-o"
        size="xlarge"
    />
</div>
```

2. Icon Custom Colors:

```js
<div className="icon-wrapper">
    <Icon
        name="github"
        size="xsmall"
        color="#454d5d"
    />
    <Icon
        name="github"
        size="small"
        color="#32b643"
    />
    <Icon
        name="github"
        size="medium"
        color="#ffb700"
    />
    <Icon
        name="github"
        size="large"
        color="#e85600"
    />
    <Icon
        name="github"
        size="xlarge"
        color="#e06870"
    />
</div>
```