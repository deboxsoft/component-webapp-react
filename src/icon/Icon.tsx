import React from 'react';
import styled, { css } from 'styled-components/macro';
import { OnClickedProps, StyledThemeProps } from '../types';
import { IconSize } from './types';
import { IconTheme } from '../theme';

interface IconStyledProps extends StyledThemeProps<IconTheme> {
  size: keyof IconSize;
  color?: string;
  className?: string;
}

interface IconProps extends IconStyledProps, OnClickedProps {
  /**
   * The name of the icon - refer font awesome for list of supported icons
   */
  name: string;
}

type DefaultProps = IconStyledProps;

const IconStyled = styled.i.attrs(({ color }) => ({
  style: color && {
    color
  }
}))<IconStyledProps>`
  position: relative;
  cursor: pointer;
  text-align: center;

  ${props => {
    const { theme, size } = props;
    return (
      size &&
      css`
        height: ${theme.icon.height && theme.icon.height[size]};
        width: ${theme.icon.width && theme.icon.width[size]};
        font-size: ${theme.icon.fontSize && theme.icon.fontSize[size]};
      `
    );
  }} ${props =>
    props.color &&
    css`
      color: ${props.color};
    `};
`;

/**
 * This component will render font awesome icons only, include font awesome in your project.
 *
 * @example ./../../docs/components/Icon.md
 */
export const Icon = ({ name, className, theme: themeProps, ...props }: IconProps) => {
  return <IconStyled {...props} className={`fa fa-${name} ${className || ''}`} aria-hidden="true" />;
};

Icon.defaultProps = {
  size: 'default',
  theme: { icon: {} }
};
