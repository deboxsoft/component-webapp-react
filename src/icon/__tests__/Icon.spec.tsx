import React from 'react';
import { render, fireEvent, cleanup, waitForElement, debugDOM } from '@testing-library/react';
import { Icon } from '../Icon';

describe('IconComponent', () => {
  afterEach(cleanup);
  it('render no error', () => {
    const { getByTestId, container } = render(<Icon data-testid="icon" name="smile-o" size="xs" />);
    const element = getByTestId('icon');
    expect(element.tagName).toBe('I');
    expect(container).toMatchSnapshot();
  });
});
