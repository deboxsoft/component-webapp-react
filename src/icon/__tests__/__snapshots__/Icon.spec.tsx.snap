// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent render no error 1`] = `
<div>
  <i
    aria-hidden="true"
    class="fa fa-smile-o  Icon__IconStyled-biexjw-0 dQVOQN"
    data-testid="icon"
  />
</div>
`;
